src/main/java-templates/org/postgresql/Driver.java

```diff
@@ -411,7 +411,7 @@ public Connection getResult(long timeout) throws SQLException {
      * @throws SQLException if the connection could not be made
      */
     private static Connection makeConnection(String url, Properties props) throws SQLException {
-        return new @JDBCCONNECTCLASS@(hostSpecs(props),
+        return new ${JDBCCONNECTCLASS}(hostSpecs(props),
```

Comment:
Yes, it's easier to understad having a piece of code to lookt at. :-)

I think we should separate discussions here:
* How to do preprocessing in Maven?
* Is preprocessing a good thing at all?

The target of the current PR is Mavenization. Hence, the question is "How to do preprocessing in Maven?". That's pretty simple: Maven can filter source code automatically before compilation, so the solution as-is will work once Maven is correctly configured. What you need is called the "Templating Maven Plugin", which is an alpha currently and the web site is moving from Codehaus to Mojohaus at the moment (bad, so you cannot do it NOW). A different approach is to apply a workaround using a combination of the "Maven Resources Plugin" (which can filter pretty anything as long as it is text) and setting the source path to the output of that plugin. We use that a lot that way to patch version numbers and build dates into the source code and it works pretty well. The drawback is that least IDEs any of these solutions and won't be able to support it. So our conclusion is that we do not like preprocessing as it makes things complex.

The second question is whether it is a good idea to patch sources at all. I do not think this should be part of the same PR, but we should first open an issue for that and discuss in a larger audience. My personal opinion is that I would keep one single source code for one single platform. It makes development pretty easy and new people can jump right in easily. I'd call that "one code for one platform". If really multiple platforms or even a matrix shall be supported (and it looks like the project leads want that), I'd go with git branches. A different solution would be using the Strategy pattern, hence solving the complexity at runtime by simply loading different implementations.

In fact, the problem is less the different JDBC levels. We could simply always support the latest level, as the JVM will not complain about unkown classes until we really instantiate them, which is not the case if the code is kept JDK 6 compliant (you can offer interface that uses JDK 8 without the need to load JDK 8 classes until that class is really used). Maybe this is the way you like to go?